agent_integration_test.go

```diff
@@ -0,0 +1,64 @@
+//go:build integration
+// +build integration
+
+package tfe
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAgentsRead(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	agent, _, agentCleanup := createAgent(t, client, nil, nil, nil)
+	defer agentCleanup()
+	t.Log("log agent: ", agent)
+	// t.Log("log pool: ", pool)
+
+	t.Run("when the agent exists", func(t *testing.T) {
+		k, err := client.Agents.Read(ctx, agent.ID)
+		require.NoError(t, err)
+		assert.Equal(t, agent, k)
+	})
+
+	t.Run("when the agent does not exist", func(t *testing.T) {
+		k, err := client.Agents.Read(ctx, "nonexisting")
+		assert.Nil(t, k)
+		assert.Equal(t, err, ErrResourceNotFound)
+	})
+
+	t.Run("without a valid agent ID", func(t *testing.T) {
+		k, err := client.Agents.Read(ctx, badIdentifier)
+		assert.Nil(t, k)
+		assert.EqualError(t, err, ErrInvalidAgentID.Error())
+	})
+}
+
+func TestAgentsList(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	agent, agentPool, agentCleanup := createAgent(t, client, nil, nil, nil)
```

Comment:
@sebasslash I'm curious what you think here. Per your recommendation not to return *AgentPool in the createAgent helper fn, when you get here, you want to pass an agent pool that is connected to the agent created in the helper. Does that make sense?

Reply:
Nvm! We chatted!